src/hotspot/share/gc/g1/g1ServiceThread.cpp

```diff
@@ -130,6 +131,12 @@ void G1ServiceThread::run_task(G1ServiceTask* task) {
 
   task->execute();
 
+  if (UsePerfData && os::is_thread_cpu_time_supported()) {
+    CPUTimeCounters* counters = G1CollectedHeap::heap()->cpu_time_counters();
+    ThreadTotalCPUTimeClosure tttc(counters, CPUTimeGroups::gc_service);
+    tttc.do_thread(task->_service_thread);
+  }
```

Comment:
Please extract this to a function, similar to the other cases something like `track_cpu_time()`.